### ISSUE TITLE
Ability to "negate" rules, and the use of OR combinations

### ISSUE THREAD

**Issue by cbuijs** [folbricht/routedns#28]
For now it is only possible to create (router) rules that are 'HIT' but not when not-hit, or an else case if you like.

Example, it would be nice for all rules to add a exclaimation-mark/point in front of the definition to 'negate' the rule, making it a 'not' rule.

Example (router rule):
`{ type = "!A", resolver = "null-a" },`

The `!A` would mean "when NOT" an `A` type.

This probably will imply to make rules more flexible and have `OR` combo's to simplify the ruling, for example:
`{ type = "!CNAME,A,AAAA", resolver = "null-a" },`

Meaning "when NOT a `CNAME` or IS an `A` or IS an `AAAA`", it would be sequential and first hit wins in order.

Would be great if we could adapt this to domains/hosts as well to have a more granular "exclude" facility within block/allow. For example, allow `.google.com` except `badstuff.google.com`.

There is maybe some logic that needs to be changed just to make sure that things are mutual exclusive, which I think is just ordering things right in the config and document it. Logging maybe needs to be enriched on the "negated hit" info as well.

**Comment by folbricht** [comment]
There's now a first implementation of this on the issue-28 branch if you want to give it a try. It adds two things:

1) Multiple types in one route. Note the `s` in `types`. The old `type` attribute is still supported.

`{ name = '(^|\.)google\.com\.$', types = ["A", "AAAA", "MX"], resolver="google-dot" },`

2) Inverting of routes. The `invert=true` attribute will invert the result of the whole route. So you can do things like this:

```toml
[routers.router1]
routes = [
  { invert = true, types = ["A", "AAAA", "MX"], resolver="static-nxdomain" }, # disallow anything that is not A, AAAA, or MX
  { resolver="cloudflare-dot" }, # default route
]
```

**Comment by cbuijs** [comment]
So far, so good. Works as expected and made my routers sections more manageable, sweet!

Adding the `invert` parameter also to the groups rules would be grant as well.

PS: You might want to consider to simplify the allow/block-list construction of blocklist-v2 (and the other blocklist-types), to have a `non-hit-resolver` and `hit-resolver` instead of having allowlist/blocklist ones to make it more abstract.

**Comment by folbricht** [comment]
Merged in #97
